pkg/controller/clusterdeployment/clusterinstalls.go

```diff
@@ -151,6 +151,7 @@ func (r *ReconcileClusterDeployment) reconcileExistingInstallingClusterInstall(c
 		statusModified = true
 	}
 
+	completed = controllerutils.FindClusterDeploymentCondition(conditions, hivev1.ClusterInstallCompletedClusterDeploymentCondition)
```

Comment:
I can't quite tell, is there a reason why we're finding this condition again when it was done a few lines above? Or was this a mistaken paste?

Reply:
Since `FindClusterDeploymentCondition` returns a pointer to the condition in question from the list of conditions, when we update the list of conditions (the list is updated right above this code - it is setting a condition that in turn also sorts it), the pointer changes - so we need to re-fetch the accurate pointer. Caught by a failing unit-test

Reply:
Wow, would never have thought. Thanks!

Reply:
can we return a deepcopy or something ?? this is not great to remember(hey after update your previous reads might be invalid...)?